syntax = "proto3";
package notify_push;

message RegisterTokenRequest {
  string device_id = 1;
  string device_type = 2;
  string token = 3;
}

message BaseResponse {
    bool success = 1;
}

message ServerResponse {
    StunServer stun_server = 1;
    TurnServer turn_server = 2;
}

message StunServer {
  string server = 1;
  int64 port = 2;
}

message TurnServer {
  string server = 1;
  int64 port = 2;
  string type = 3;
  string user = 4;
  string pwd = 5;
}

// Method
service NotifyPush {
    rpc register_token (RegisterTokenRequest) returns (ServerResponse);
}
